src/game/client/components/chat.cpp

```diff
@@ -410,6 +417,17 @@ bool CChat::OnInput(IInput::CEvent Event)
 		}
 	}
 
+	//Handle Chat Buffer
+	if(Event.m_Flags&IInput::FLAG_PRESS && (Event.m_Key == KEY_RETURN || Event.m_Key == KEY_KP_ENTER))
+	{
+		ClearChatBuffer();
+	}
+	else if(Event.m_Key != KEY_MOUSE_1 && Event.m_Key != KEY_MOUSE_2)
+	{
+		//Save Chat Buffer
+		m_ChatBufferMode = m_Mode;
+		str_copy(m_ChatBuffer, m_Input.GetString(), m_Input.GetLength()+1);
```

Comment:
3rd parameter should be size of m_ChatBuffer

Reply:
okay